homeassistant/components/zha/__init__.py

```diff
@@ -285,13 +301,47 @@ def register_entity(self, ieee, entity_obj):
         discovery_info.update(entity_info)
         self._hass.data[DISCOVERY_KEY][cluster_key] = discovery_info
 
-        await discovery.async_load_platform(
-            self._hass,
-            component,
-            DOMAIN,
-            {'discovery_key': cluster_key},
-            self._config,
-        )
+        from zigpy.zcl.clusters.general import OnOff
+        if component == 'binary_sensor' and OnOff.cluster_id == \
+                cluster.cluster_id:
```

Comment:
Should we better add something like `SINGLE_OUTPUT_CLUSTER_EVENT_CLASS` to zha\const.py and handle those in a separate pass from handling SINGLE_INPUT/OUTPUT_DEVICE_CLASSes??? in current implementation, a "hypothetical" IAS device with an OnOff input cluster would cause the event handler to be instantiated, even if there's no OnOff output cluster, wouldn't it?

Reply:
Yes, was speaking to @damarco about this as well. I am going to fix this. I ordered some Xiaomi sensors to test this case specifically because apparently they work this way. I have to add a sensor class back into binary_sensor to fix this as well.